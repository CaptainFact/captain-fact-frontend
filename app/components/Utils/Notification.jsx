import React from 'react'

const Notification = ({ children, type = 'info', ...otherProps }) => (
  <div className={`notification is-${type}`} {...otherProps}>
    {children}
  </div>
)

export default Notification
