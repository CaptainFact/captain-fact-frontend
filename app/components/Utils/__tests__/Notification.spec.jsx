import Notification from '../Notification'

test("render message", () => {
  snapshot(<Notification>Hello</Notification>)
})

test("can set type", () => {
  snapshot(<Notification type="warning">Alert !</Notification>)
})

test("pass props", () => {
  snapshot(<Notification title="my title">Hellooow</Notification>)
})

test("without body", () => {
  snapshot(<Notification/>)
})
