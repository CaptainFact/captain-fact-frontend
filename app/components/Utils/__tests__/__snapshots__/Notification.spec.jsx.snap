// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can set type 1`] = `
<div
  className="notification is-warning"
>
  Alert !
</div>
`;

exports[`pass props 1`] = `
<div
  className="notification is-info"
  title="my title"
>
  Hellooow
</div>
`;

exports[`render message 1`] = `
<div
  className="notification is-info"
>
  Hello
</div>
`;

exports[`without body 1`] = `
<div
  className="notification is-info"
/>
`;
